dash/src/memory/internal/MemorySpaceRegistry.cc

```diff
@@ -0,0 +1,72 @@
+#include <algorithm>
+
+#include <dash/Exception.h>
+#include <dash/internal/Logging.h>
+#include <dash/memory/internal/MemorySpaceRegistry.h>
+
+std::unique_ptr<dash::internal::MemorySpaceRegistry>
+    dash::internal::MemorySpaceRegistry::m_instance;
+
+std::vector<dash::internal::MemorySpaceRegistry::element_t>
+    dash::internal::MemorySpaceRegistry::m_segments;
+
+std::once_flag dash::internal::MemorySpaceRegistry::m_onceFlag;
+
+std::ostream &operator<<(std::ostream &os, const dart_gptr_t &dartptr);
+
+namespace dash {
+namespace internal {
+MemorySpaceRegistry& MemorySpaceRegistry::GetInstance()
+{
+  std::call_once(
+      m_onceFlag, [] { m_instance.reset(new MemorySpaceRegistry); });
+  return *m_instance.get();
+}
+
+bool MemorySpaceRegistry::add(dart_gptr_t p, value_t value)
+{
+  auto key = std::make_pair(p.teamid, p.segid);
+
+  if (do_lookup(key) != std::end(m_segments)) {
+    return false;
+  }
+
+  DASH_LOG_TRACE(
+      "MemorySpaceRegistry.add", "adding memory space segment", p, value);
+
+  m_segments.emplace_back(std::make_pair(key, value));
+
+  return true;
+}
+
+void MemorySpaceRegistry::erase(dart_gptr_t p)
+{
+  auto const key = std::make_pair(p.teamid, p.segid);
+
+  auto it = do_lookup(key);
+
+  if (it == std::end(m_segments)) {
+    return;
+  }
+
+  DASH_LOG_TRACE(
+      "MemorySpaceRegistry.erase",
+      "removing memory space",
+      std::get<IDX_TEAMID>(key),
+      std::get<IDX_SEGID>(key));
+
+  m_segments.erase(it);
+}
+
+std::vector<dash::internal::MemorySpaceRegistry::element_t>::iterator
```

Comment:
Remove `dash::internal` from `dash::internal::MemorySpaceRegistry::element_t`

Reply:
Ok should be done now.